jenkins/scripts/dynamic_worker_workflow/e2e_tests.sh

```diff
@@ -39,24 +39,33 @@ if [[ "${CAPM3_VERSION}" == "v1alpha5" ]]; then
     export KUBERNETES_VERSION="v1.23.8"
     export KUBECTL_SHA256="${KUBECTL_SHA256:-4685bfcf732260f72fce58379e812e091557ef1dfc1bc8084226c7891dd6028f}"
 fi
-if  [[ -z "${KUBERNETES_VERSION:-}" ]]; then
+
+if [[ "${NUM_NODES}" == "null" ]]; then
```

Comment:
```suggestion
if [[ -z "${NUM_NODES:-}" ]] || [[ "${NUM_NODES}" == "null" ]]; then
```
Better to have these all also not explode on undefined values.

Reply:
agreed